### PR TITLE
[FLINK-21717][docs] Fix broken links to Java and Python Docs

### PR DESCRIPTION
**Pull request by LadyForest**
## What is the purpose of the change

This PR is to fix the broken links to Java/Python docs since after migrating to Hugo, the old references became unreachable.

## Brief change log

  - Fix the typo in `javadoc.html`.
  - Add shortcode for Python docs -> `pythondoc.html` 
  - Replace and fix references in docs.

## Verifying this change

Rebuild the docs will verify the fix.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/layouts/shortcodes/pythondoc.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/}}{{/*
+    Shortcode for linking to Flink's PythonDoc
+
+    Parmeters:
+        - name: The rendered link name (required)
```
Comment (zentol) [inline docs/layouts/shortcodes/pythondoc.html:22]: We could also document the file parameter

File: docs/layouts/shortcodes/javadoc.html [hunk 2]
```diff
@@ -16,11 +16,11 @@
...
-    Shortcode for linking to Flinks JavaDoc
+    Shortcode for linking to Flink's JavaDoc
 
     Parmeters: 
         - name: The rendered link name (required)
 */}}
-<a href="{{ .Site.Params.JavaDoc }}/{{ .Get "file" }}">
-    {{ .Get "name" }}
+<a href="{{ .Site.Params.JavaDocs }}/{{ .Get "file" }}">
+    {{ .Get "name" | markdownify }}
```
Comment (zentol) [inline docs/layouts/shortcodes/javadoc.html:25]: what is the goal here?

**Review by zentol — CHANGES_REQUESTED** [review]
Could you open a JIRA ticket for this PR? I believe it is large enough to warrant one.

File: docs/layouts/shortcodes/javadoc.html (hunk 2 above)
Comment (LadyForest) [reply docs/layouts/shortcodes/javadoc.html:25]: > Could you open a JIRA ticket for this PR? I believe it is large enough to warrant one.

Sure. https://issues.apache.org/jira/browse/FLINK-21717

Comment (LadyForest) [reply docs/layouts/shortcodes/javadoc.html:25]: > what is the goal here?

`JavaDoc` is a typo. The params defined in config.toml is `JavaDocs`.

Comment (zentol) [reply docs/layouts/shortcodes/javadoc.html:25]: And what about `markdownify`? As in, what problems existed without it?

Comment (LadyForest) [reply docs/layouts/shortcodes/javadoc.html:25]: `{{ page.excerpt | markdownify }}` is illustrated in https://jekyllrb.com/docs/liquid/filters/ that converts a Markdown-formatted string into HTML.  See also `docs/layouts/shortcodes/gh_link.html`.

**Pull request merged**